src/sentry/testutils/helpers/task_runner.py

```diff
@@ -9,11 +9,12 @@
 
 @contextmanager
 def TaskRunner():
+    prev = settings.CELERY_ALWAYS_EAGER
     settings.CELERY_ALWAYS_EAGER = True
     current_app.conf.CELERY_ALWAYS_EAGER = True
     yield
-    current_app.conf.CELERY_ALWAYS_EAGER = False
-    settings.CELERY_ALWAYS_EAGER = False
+    current_app.conf.CELERY_ALWAYS_EAGER = prev
+    settings.CELERY_ALWAYS_EAGER = prev
```

Comment:
Did we have nested taskrunner contexts?

Reply:
Yes, we did, unfortunately.  This fixed subtle bugs as a result of that.